### PR TITLE
Jimmy Jurado

### PR DESCRIPTION
**Pull request by jcjurado3**
Started off rocky when creating the new tables and associations. I reviewed the many to many lesson right before as this is one stuggle I had but that really helped in clarifying the join table and the need for it. Had syntax error to start but just followed the error message and referenced the set list repo and the many to many lesson plan. My errors were missing s or simply adding a do when I didn't need it. 

After reviewing my mistakes on the first IC, I understood my error and corrected it. I also attended the IC review which cleared things up and understood my confusion with associations. 

I was proud of completing Ext. 1 as I do believe I have it set up correctly. Prys came in handy and could locate the actors inside the movies and the movies inside the studios so then it was a matter of iterating properly. Took two tries as I attempted to do studio.movies.actor but after prying a bit I knew I had to add another iteration. I did tag a distinct as it is presentation logic. Creating a method for this didn't seem right or needed. 

Overall, I still had to reference the practice repos to get started but felt confident in my actions and placement of model instance methods, keeping the controller clean, and call the method on the view.

### CODE REVIEW

File: app/controllers/movie_actors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class MovieActorsController < ApplicationController
+  def create
+    movie = Movie.find(params[:movie_id])
+    movie_actor = MovieActor.create!(mov_act_params)
+
+    redirect_to "/movies/#{movie.id}"
+  end
+
+  private
+  def mov_act_params
```
Comment (epintozzi) [inline app/controllers/movie_actors_controller.rb:10]: This is probably more a personal opinion of mine that any firm thing that needs to change, but I don't love abbreviations in method and variable names. This is a small app so it's easy to figure out what this means. Once apps start getting bigger and more complex, it's a lot more mental load keeping track of things that are unnecessarily ambiguous. I'd probably name this `movie_actor_params` and just spell the whole thing out.

File: app/controllers/studios_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class StudiosController < ApplicationController
+  def index
+    @studio = Studio.all
+
```
Comment (epintozzi) [inline app/controllers/studios_controller.rb:4]:
```suggestion
```

File: app/controllers/studios_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class StudiosController < ApplicationController
+  def index
+    @studio = Studio.all
+    
+  end
+
+  def show
+    @studio = Studio.find(params[:id])
+
```
Comment (epintozzi) [inline app/controllers/studios_controller.rb:9]:
```suggestion
```

File: app/models/actor.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require "rails_helper"
```
Comment (epintozzi) [inline app/models/actor.rb:1]: We should only be requiring rails_helper in our specs. If there's code you've added to your helper specific to application code, there is probably a better place for it to live.

File: app/models/movie_actor.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
```
Comment (epintozzi) [inline app/models/movie_actor.rb:1]: See comment above
```suggestion
```

**Review by epintozzi — COMMENTED** [review]
Great work!

File: app/controllers/movie_actors_controller.rb (hunk 1 above)
Comment (jcjurado3) [reply app/controllers/movie_actors_controller.rb:10]: Thanks for this note!

File: app/models/actor.rb (hunk 4 above)
Comment (jcjurado3) [reply app/models/actor.rb:1]: Makes sense! Went rails_helper happy

**Pull request closed without merging**